aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,1719 @@
+        let (
+            lockup_cycle_ended,
+            active,
+            pending_inactive,
+            commission_active,
+            commission_pending_inactive
+        ) = calculate_stake_pool_drift(pool);
+
+        active = pool_u64::shares_to_amount_with_total_coins(
+            &pool.active_shares,
+            pool_u64::shares(&pool.active_shares, delegator_address),
+            // exclude operator active rewards not converted to shares yet
+            active - commission_active
+        );
+
+        // if no pending withdrawal, there is neither inactive nor pending_inactive stake
+        let (withdrawal_exists, withdrawal_lockup_cycle) = pending_withdrawal_exists(pool, delegator_address);
+        let inactive;
+        (inactive, pending_inactive) = if (withdrawal_exists) {
+            // delegator has either inactive or pending_inactive stake due to automatic withdrawals
+            let inactive_shares = table::borrow(&pool.inactive_shares, withdrawal_lockup_cycle);
+            if (withdrawal_lockup_cycle < pool.observed_lockup_cycle) {
+                // if withdrawal's lockup cycle was ended on delegation pool then its stake is inactive
+                (pool_u64::balance(inactive_shares, delegator_address), 0)
+            } else {
+                pending_inactive = pool_u64::shares_to_amount_with_total_coins(
+                    inactive_shares,
+                    pool_u64::shares(inactive_shares, delegator_address),
+                    // exclude operator pending_inactive rewards not converted to shares yet
+                    pending_inactive - commission_pending_inactive
+                );
+                // if withdrawal's lockup cycle was ended ONLY on stake pool then its stake is inactive
+                if (lockup_cycle_ended) { (pending_inactive, 0) } else { (0, pending_inactive) }
+            }
+        } else { (0, 0) };
+
+        // should also include commission rewards in case of the operator account
+        // operator rewards are actually used to buy shares which is introducing
+        // some imprecision (received stake would be slightly less)
+        // but adding rewards onto the existing stake is still a good approximation
+        if (delegator_address == stake::get_operator(pool_address)) {
+            active = active + commission_active;
+            if (lockup_cycle_ended) {
+                inactive = inactive + commission_pending_inactive;
+            } else {
+                pending_inactive = pending_inactive + commission_pending_inactive;
+            }
+        };
+        (active, inactive, pending_inactive)
+    }
+
+    #[view]
+    /// Return the unique observed lockup cycle where delegator `delegator_address` may have
+    /// unlocking (or already unlocked) stake to be withdrawn from delegation pool at `pool_address`.
+    public fun get_pending_withdrawal(pool_address: address, delegator_address: address): (bool, u64) acquires DelegationPool {
```

Comment:
nit: Rename this to has_pending_withdrawal so we can have a separate get_withdrawal or get_pending_withdrawal as discussed in an earlier comment.